File: app/src/main/java/br/com/ifood/tweetmood/presentation/di/AppModule.java
package br.com.ifood.tweetmood.presentation.di;

import android.app.Application;
import android.content.Context;

import javax.inject.Singleton;

import br.com.ifood.tweetmood.presentation.view.getuser.di.GetUserActivityComponent;
import br.com.ifood.tweetmood.presentation.view.tweets.di.MainTwitterActivityComponent;
import dagger.Module;
import dagger.Provides;

/**
 * Created by uchoa on 16/06/18.
 */

@Module(subcomponents = {
        GetUserActivityComponent.class,
        MainTwitterActivityComponent.class,
})
public class AppModule {
    @Provides
    @Singleton
    Context providesContext(Application application){
        return application;
    }
}

File: app/src/main/java/com/rlino/ifoodtwitterchallenge/di/AppModule.kt
package com.rlino.ifoodtwitterchallenge.di

import android.content.Context
import com.rlino.ifoodtwitterchallenge.App
import dagger.Module
import dagger.Provides

@Module
class AppModule {

    @Provides
    fun provideContext(application: App): Context {
        return application.applicationContext
    }

}